cli/internal/publish/pluginui.go

```diff
@@ -65,14 +81,35 @@ func UploadPluginUIAssets(ctx context.Context, c *cloudquery_api.ClientWithRespo
 	eg, egCtx := errgroup.WithContext(ctx)
 	eg.SetLimit(4)
 
+	var bundleAsset cloudquery_api.PluginUIAsset
 	for _, asset := range resp.JSON201.Assets {
+		if asset.Name == uiAssetBundleTarName {
+			bundleAsset = asset
+			continue
+		}
+
 		asset := asset
 		details := urlPathVsDetails[asset.Name]
 		eg.Go(func() error {
 			return hub.UploadFileWithContentType(egCtx, asset.UploadURL, details[0], details[1])
 		})
 	}
 
+	if bundleAsset.UploadURL == "" {
+		return errors.New("bundle asset URL not found in the response")
+	}
+
+	bundleReader, bundleWriter := io.Pipe()
+	defer bundleWriter.Close()
+
+	eg.Go(func() error {
```

Comment:
Should we be re-using the same error group here? Asking since it's limited to 4 Go routines so we might be writing to the writer without a Go routine that reads from it (at least until one is available).
Don't think it can hang but might be better to have a dedicated error group with 2 Go routines, one for writing, another for reading

Reply:
Refactored into a separate function w/ a nested errgroup [c1a544e](https://github.com/cloudquery/cloudquery/pull/20116/commits/c1a544edd411fe64d963c0497829a127f9504ff4)